Java/dp/Rod_Cutting.java

```diff
@@ -0,0 +1,105 @@
+/*Problem Description
+
+You are given a rod of length N inches and an array of prices that contains prices of all pieces of size smaller than N. 
+Determine the maximum value obtainable by cutting up the rod and selling the pieces.
+
+*/
+
+// importing required package 
+import java.util.Scanner;
+
+class Helper {
+	
+    // this method - getValue() will return the final answer to main method 
+    public int getValue(int price[], int n) {
+
+        // m variable will help in getting the size array length by using price length 
+        int m = price.length ;
+        int size[] = new int[m];
+        for(int i = 0 ; i < m  ; i++)
+            size[i] = i + 1;
+
+        // call solve() method for solving the problem
+        int ans = solve(price , size, m , n);
+        return ans ;
+    }
+    
+    //solve() method will act as solving the problem
+
+    int solve(int[] price , int[] size , int m , int n  )
+    {
+
+        // Create 2-D array to store the value of profit 
+        int dp[][] = new int[m+1][n+1] ;
+        
+        for(int i = 0 ; i < m + 1; i++)
+        {
+            for(int j = 0 ; j < n + 1 ; j++)
+            {
+                // if size = 0 or there is 0 value for the size
+                if(i == 0 || j ==0 )
+                {
+                    dp[i][j] = 0;
+                    
+                }
+
+                /*  if size is less than the current max size then 
+                  get the max price from size or neglect if not needed
+                  */
+                else if(size[i-1] <= j )
+                {
+                    dp[i][j] = Math.max(price[i-1] + dp[i][j - size[i-1] ] , dp[i-1][j] ) ;
+                }
+
+                // if size is greater than current max size , no value is taken from price array
+                else
+                {
+                    dp[i][j] = dp[i-1][j];
+                }
+            }
+        }
+        return dp[m][n];
+    }
+    
+}
+
+// main class which contains main method
+public class Rod_Cutting {
+	
+	 public static void main(String args[]) {
+
+	        Scanner sc = new Scanner(System.in);
+	        int test = sc.nextInt();
+	        while (test-- > 0) {
+	            int n = sc.nextInt();
+	            int[] arr = new int[n];
+
+                // get values of price 
+	            for (int i = 0; i < n; i++) 
+	            	arr[i] = sc.nextInt();
+
+            //Creating object of another class 
+	            Helper help = new Helper();
+	            System.out.println(help.getValue(arr, n));
+	        }
+	    }
+	 
```

Comment:
Remove the extra one line space

Reply:
See the final file. Applied the change .